app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -219,7 +230,8 @@ class ThreadAdapter(
         }
 
         if (item is MessageUi) {
-            (holder as MessageViewHolder).bindMail(item, position)
+            val messageCount = runCatchingRealm { items.count { it is MessageUi } }.getOrDefault(0)
```

Comment:
Here we can avoid looping over the list to count the items if we access the cached messageCount in the view model